polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisPrincipal.java

```diff
@@ -74,4 +106,8 @@ static PolarisPrincipal of(String name, Map<String, String> properties, Set<Stri
    * as permissions, preferences, or other metadata.
    */
   Map<String, String> getProperties();
+
+  /** Returns the access token of the current user. */
+  @Nullable
+  String getToken();
```

Comment:
nit: WDYT about `Optional<String>` here?

Reply:
It's a "nit" comment :) `Optional` is slightly preferable IMHO because it's an essential piece of information for making the `StorageAccessConfig`, but the content of the token is optional... really fine point :) I'm ok either way.